controller/konnect/watch_kongvault.go

```diff
@@ -0,0 +1,46 @@
+package konnect
+
+import (
+	"context"
+	"reflect"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	operatorerrors "github.com/kong/gateway-operator/internal/errors"
+
+	configurationv1alpha1 "github.com/kong/kubernetes-configuration/api/configuration/v1alpha1"
+)
+
+// KongVaultReconciliationWatchOptions returns the watch options for KongVault.
+func KongVaultReconciliationWatchOptions(cl client.Client) []func(*ctrl.Builder) *ctrl.Builder {
```

Comment:
This should also enqueue for `KonnectGatewayControlPlane` and `KonnectAPIAuthConfiguration` like other types.

Reply:
added watches.